### ISSUE TITLE
Stage or Remove the OpenStack Cloud Provider 

### ISSUE THREAD

**Issue by andrewsykim** [kubernetes/kubernetes#80041]
**What would you like to be added**:
As part of a long running effort to [remove all in-tree cloud providers](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/20190125-removing-in-tree-providers.md), we either deleted or staged the in-tree cloud providers in [k8s.io/legacy-cloud-providers](https://github.com/kubernetes/legacy-cloud-providers). OpenStack was not deleted or staged because of it's internal dep to `pkg/util/mount`. We should either continue refactoring work for `pkg/util/mount` so we can stage the open stack cloud provider as well or just remove it. 

**Why is this needed**:
See KEP [remove all in-tree cloud providers](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/20190125-removing-in-tree-providers.md).

**Comment by dims** [comment]
We should remove openstack cloud provider. @davidz627 and @ddebroy are trying to get `CSIMigration` to Beta in 1.16, So this removal can be done in 1.16 itself.

**Comment by andrewsykim** [comment]
cc @hogepodge @adisky

**Comment by dims** [comment]
Please see https://github.com/kubernetes/kubernetes/pull/67782#issuecomment-426437569 for additional context. I've been waiting for this for a while! :)

**Comment by davidz627** [comment]
@dims I'm not sure how removal can be done in `1.16` follows from `CSIMigration` to Beta in `1.16`. Are these related? The actual "removal" of in-tree plugin code from a storage CSI Migration timeline perspective is outlined in this doc: https://docs.google.com/document/d/16h9vLLbqQCN83KnlPtn7qL_GB1TYwhWkoA5G4FS3KBI/edit?usp=sharing

**Comment by dims** [comment]
@davidz627 When we have the `CSIMigration` enabled in 1.16,  folks will be able to continue using their existing cinder yaml(s) right? that was the thing that is blocking removing all the openstack related in-tree code.

**Comment by davidz627** [comment]
It will be beta and off by default. Please see doc for timelines.

TL;DR: You can't remove in-tree code until there is a viable CSI alternative that slots into the backend with `CSIMigration` with no/minimal user disruption. This requires things like driver feature parity, driver deployed by default, etc.

**Comment by adisky** [comment]
@dims We are working on the feature parity for both drivers, yet a rigorous testing needs to be done after migration flag enabled.  We are not testing every features of cinder csi driver (e2e testing in progress). Apart from that most important thing, we do not have any deployment strategy for csi driver.

**Comment by dims** [comment]
@adisky Agreed. +1 to "rigorous testing needs to be done after migration flag". yes for the "deployment strategy" we just follow what ever the CSIMigration folks come up with (or the sig-storage folks in general). it's not a openstack/cinder specific problem.

**Comment by adisky** [comment]
for reference: https://github.com/kubernetes/kubernetes/pull/80027

**Comment by hogepodge** [comment]
/remove-lifecycle stale

@adisky what's the status of the final removal? Is the Cinder driver nearly ready to go?

**Comment by adisky** [comment]
@hogepodge We need to wait for CSI migration to be GA.
